### PR TITLE
Fix "Build artifacts" job and merge with "Update mirrors"

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

When we removed .svnignore, that caused the job to start trying to
include all of node_modules and such in the artifact which is completely
not what we want.

We already have the logic to produce exactly what we do want for the
mirror repos, so merge those workflows (and separate building from pushing in the process).

#### Jetpack product discussion

p1610657658129800-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* See that the new workflow built proper artifacts.
* See that the correct artifact was pushed to Jetpack Beta.

#### Proposed changelog entry for your changes:

* None needed.

### CODE REVIEW

File: tools/build-jetpack.sh [hunk 1]
```diff
@@ -1,97 +0,0 @@
...
-    $TARGET_DIR
-
-cd $JETPACK_DIR
-
-if [[ $ADD_BETA_VERSION -eq 1 ]]; then
-    echo "Changing version of the Jetpack to reflect the latest changes.."
-    CURRENT_VERSION="`$GET_VERSION_SCRIPT`"
-
-    sed -i -e 's/Version: .*$/Version: '$CURRENT_VERSION'/' jetpack.php
-    echo $CURRENT_VERSION > version.txt
```
Comment (kraftbj) [inline tools/build-jetpack.sh:56]: We need to either bring this back OR change the "inform beta builder step"

**Review by kraftbj — CHANGES_REQUESTED** [review]

**Comment by anomiex** [comment]
Hmm. Version number pushed to Jetpack Beta isn't quite right, because it's not picking up the git version.

**Comment by kraftbj** [comment]
@anomiex and what the beta server slurped up is wrong too. https://betadownload.jetpack.me/branches/fix_build-beta-artifacts/jetpack-dev.zip

**Comment by anomiex** [comment]
Grr. It just arbitrarily slurped the first artifact instead of the one named "jetpack-dev".

**Review by kraftbj — APPROVED** [review]

**Comment by kraftbj** [comment]
Testing thread: p1610723715188300-slack-CBG1CP4EN

**Pull request merged**